src/paths/dijkstra.c

```diff
@@ -617,23 +616,35 @@ int igraph_get_shortest_path_dijkstra(const igraph_t *graph,
  * in order. Assumes that both paths are pointers to igraph_vector_t
  * objects and that they are not empty
  */
-static int igraph_i_vector_tail_cmp(const void* path1, const void* path2) {
-    return (int) (igraph_vector_tail(*(const igraph_vector_t**)path1) -
-                  igraph_vector_tail(*(const igraph_vector_t**)path2));
+static int igraph_i_vector_tail_cmp(void *extra, const void* a, const void* b) {
+    igraph_vector_ptr_t *paths = (igraph_vector_ptr_t *) extra;
+    long int *aa = (long int*) a;
+    long int *bb = (long int*) b;
+    igraph_vector_t *path_a = (igraph_vector_t *) VECTOR(*paths)[*aa];
+    igraph_vector_t *path_b = (igraph_vector_t *) VECTOR(*paths)[*bb];
+
+    return (int) (igraph_vector_tail(path_a) - igraph_vector_tail(path_b));
```

Comment:
There might be a subtle bug here (and it has been there for a while now). If `igraph_vector_tail(path_a) - igraph_vector_tail(path_b)` (which is of type `igraph_real_t`) happens to be equal to a multiple of `INT_MAX+1`, that _might_ be cast into a zero. I don't know whether it actually happens for doubles; it definitely happens when casting from `long int` to `int` and the `long int` is an exact multiple of `INT_MAX+1`, and I'm not sure how compliant C compilers are, so it's probably better to spell out the comparisons explicitly. I'll fix that in a new commit.